src/OpenFeature/Model/ImmutableMetadata.cs

```diff
@@ -45,7 +45,13 @@ public ImmutableMetadata(Dictionary<string, object> metadata)
     /// <returns>The integer value associated with the key, or null if the key is not found.</returns>
     public int? GetInt(string key)
     {
-        return this.GetValue<int>(key);
+        var hasValue = this._metadata.TryGetValue(key, out var value);
+        if (!hasValue)
+        {
+            return null;
+        }
+
+        return value is double || value is int ? Convert.ToInt32(value) : null;
```

Comment:
Just a quick thought - do we want to allow implicit conversion here (e.g., from `double`) using `Convert.ToInt32`? Since it can silently round or truncate, I’m curious if that’s acceptable for our use case.

Would it make sense to add an explicit check like this to avoid unexpected rounding?

```csharp
if (value is double d && d % 1 == 0)
    return Convert.ToInt32(d);
```